### ISSUE TITLE
[RFE] DexBackedDexFile

### ISSUE THREAD

**Issue by Lanchon** [JesusFreke/smali#447]
add this to DexBackedDexFile:
```
    public static void verifyMagicAndByteOrder(@Nonnull InputStream is)
            throws IOException {
        if (!is.markSupported()) {
            throw new IllegalArgumentException("InputStream must support mark");
        }
        is.mark(44);
        byte[] partialHeader = new byte[44];
        try {
            ByteStreams.readFully(is, partialHeader);
        } catch (EOFException ex) {
            throw new NotADexFile("File is too short");
        } finally {
            is.reset();
        }

        verifyMagicAndByteOrder(partialHeader, 0);
    }
```

then use it from:
- DexBackedDexFile.fromInputStream
- ZipDexContainer.isDex

and also make this public:
- DexBackedDexFile.verifyMagicAndByteOrder

rationale: avoid having to read a complete dex file to recognize one; checking the header is useful oprtation for clients.

**Comment by JesusFreke** [comment]
Implemented in 4eefe294e4c664577982283cc64c415819a30c1e

**Comment by Lanchon** [comment]
thanks! and getVersion() is welcome too
